### ISSUE TITLE
java.lang.NumberFormatException in (DefaultAudioDevice.java:215)

### ISSUE THREAD

**Issue by nguyen-vinasource** [opentok/opentok-android-sdk-samples#182]
I got this crash via Fabric. It works well on my devices, but occurred with user on production, so I don't know much detail. I can only provide the log and the device info: 
 
![image](https://user-images.githubusercontent.com/20634330/47063318-c4e76000-d204-11e8-8967-eb9cc9659c27.png)

![image](https://user-images.githubusercontent.com/20634330/47063333-d03a8b80-d204-11e8-9ab0-a4e91032eefc.png)

My implementation: 
![image](https://user-images.githubusercontent.com/20634330/47063361-ed6f5a00-d204-11e8-8435-9cd41f2dc606.png)

Please help!

**Comment by marinaserranomontes** [comment]
Hi @nguyen-vinasource ,

What is the sdk version you are using? What is the sample you are testing?

Best,
Marina

**Comment by nguyen-vinasource** [comment]
Thanks @marinaserranomontes 

I am using version 2.12.1. Is it too out of date?
Oh, I didn't test the sample. This is my implementation. In my last photo, I were trying to build a session, then the log said it crash here. 

Let me update to latest version and update to you later..
Thanks again

**Comment by marinaserranomontes** [comment]
Hi @nguyen-vinasource ,

Did it work?

Best,
Marina

**Comment by marinaserranomontes** [comment]
Hi @nguyen-vinasource ,

I’m closing this issue because it has been inactive for a few weeks.

Please reopen if you still encounter this issue. You can also contribute directly to samples by providing a patch.

Thank you!

Best,
Marina

**Comment by sschendel** [comment]
I see occasional app crashes with similar exception. All Pixel devices running Android 9. 

OT version 2.15.3.

e.g.
```
java.lang.NumberFormatException: s == null
	at java.lang.Integer.parseInt(Integer.java:577)
	at java.lang.Integer.parseInt(Integer.java:650)
	at com.opentok.android.DefaultAudioDevice.<init>(DefaultAudioDevice.java:389)
	at com.opentok.android.AudioDeviceManager.initializeDefaultDevice(AudioDeviceManager.java:60)
	at com.opentok.android.Publisher.<init>(Publisher.java:468)
	at com.opentok.android.Publisher$Builder.build(Publisher.java:429)
	at com.varsitytutors.tutoringtools.av.OpenTokAudioVisualService.buildNewPublisher(OpenTokAudioVisualService.java:128)
	at com.varsitytutors.tutoringtools.av.OpenTokAudioVisualService.<init>(OpenTokAudioVisualService.java:121)
	at com.varsitytutors.tutoringtools.internal.TutoringToolsModule.providesOpenTokAudioVisualService(TutoringToolsModule.java:932)
```

**Comment by mostafahashim** [comment]
any update for this error, I can't create session in oreo

**Comment by sunweiyang** [comment]
I was able to reproduce this with a Pixel device that is experiencing this audio issue:
https://support.google.com/pixelphone/forum/AAAAb4-OgUsKdeLHq3G4PM/?hl=by

Pixel devices with this problem would fail to successfully conduct video + audio recordings of any kind. However, I've noticed that existing video chat apps fail gracefully without crashing (Messenger, Meet, Zoom). @marinaserranomontes Could OpenTok bubble up an event to be handled in a session event handler, instead of crashing?
